src/sockperf.cpp

```diff
@@ -2801,11 +2855,20 @@ int sock_set_multicast(int fd, struct fds_data *p_data) {
             }
             break;
         case AF_INET6:
-            //TODO; add IPv6 multicasting suport
+            log_err("For IPv6 - input must be an integer");
+            rc = SOCKPERF_ERR_UNSUPPORTED;
             break;
         }
     }
 
+    if (!rc && s_user_params.tx_mc_if_ix_specified) {
+        uint32_t if_ix = s_user_params.tx_mc_if_ix;
+        if (setsockopt(fd, IPPROTO_IPV6, IPV6_MULTICAST_IF, &if_ix, sizeof(int)) < 0) {
```

Comment:
This code is probably not correct because `sizeof(int)` may differ from `sizeof(uint32)`

Reply:
Agree, but it looks as minor issue

Reply:
probably you mean the opposite: sizeof(uint32_t)==4, sizeof(int) may vary [from 2 to 4](https://en.cppreference.com/w/cpp/language/types#Properties)